codec/src/main/java/io/netty/handler/codec/DelimitedLengthFieldBasedFrameDecoder.java

```diff
@@ -0,0 +1,160 @@
+        }
+        if (!delimiter.isReadable()) {
+            throw new IllegalArgumentException("empty delimiter");
+        }
+
+        this.delimiter = delimiter.slice(delimiter.readerIndex(), delimiter.readableBytes());
+        this.lengthFieldCharset = lengthFieldCharset;
+
+        this.lengthAdjustment = lengthAdjustment;
+        this.trimLengthString = trimLengthString;
+    }
+
+    /**
+     * Returns the number of bytes between the readerIndex of the haystack and
+     * the first needle found in the haystack.  -1 is returned if no needle is
+     * found in the haystack.
+     */
+    private static int indexOf(ByteBuf haystack, ByteBuf needle) {
+        return DelimiterBasedFrameDecoder.indexOf(haystack, needle);
+    }
+
+    /**
+     * Create a frame out of the {@link ByteBuf} and return it.
+     *
+     * @param ctx the {@link ChannelHandlerContext} which this {@link ByteToMessageDecoder} belongs to
+     * @param in the {@link ByteBuf} from which to read data
+     *
+     * @return frame           the {@link ByteBuf} which represent the frame or {@code null} if no frame could be
+     * created.
+     */
+    @Override
+    protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
+        if (discardingTooLongFrame) {
+            long bytesToDiscard = this.bytesToDiscard;
+            int localBytesToDiscard = (int) Math.min(bytesToDiscard, in.readableBytes());
+            in.skipBytes(localBytesToDiscard);
+            bytesToDiscard -= localBytesToDiscard;
+            this.bytesToDiscard = bytesToDiscard;
+
+            failIfNecessary(false);
+        }
+
+        if (consumingLength) {
+            int delimIndex = indexOf(in, delimiter);
+            if (delimIndex < 0) {
+                return null;
+            }
+
+            final String lengthStr = in.toString(in.readerIndex(), delimIndex, lengthFieldCharset);
+            try {
+                frameLength = Integer.parseInt(trimLengthString? lengthStr.trim() : lengthStr);
+            } catch (NumberFormatException e) {
+                throw new CorruptedFrameException(
+                        String.format(
+                                "Invalid length field decoded (in %s charset): %s",
+                                lengthFieldCharset.name(),
+                                lengthStr
+                        ),
+                        e
+                );
+            }
+
+            if (frameLength < 0) {
+                throw new CorruptedFrameException("negative pre-adjustment length field: " + frameLength);
+            }
+
+            frameLength += lengthAdjustment;
+
+            //consume length field and delimiter bytes
+            in.skipBytes(delimIndex + delimiter.capacity());
+
+            //consume delimiter bytes
+            consumingLength = false;
+        }
+
+        if (frameLength > maxFrameLength) {
+            long discard = frameLength - in.readableBytes();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this substraction operation to a "long". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2184)